jetstream.go

```diff
@@ -139,11 +139,11 @@ type AckOption func(opts *jsOpts) error
 // PublishOption configures publishing messages
 type PublishOption func(opts *jsOpts) error
 
-// SubscribeOption configures JetStream subscriber behavior
-type SubscribeOption func(opts *jsOpts) error
+// ConsumerOption configures JetStream consumer behavior
+type ConsumerOption func(opts *jsOpts) error
 
-// SubscribeConsumer creates a JetStream consumer on stream based on DefaultConsumer and modified by opts
-func SubscribeConsumer(stream string, cfg *ConsumerConfig) SubscribeOption {
+// Consumer creates a JetStream Consumer on stream based on DefaultConsumer and modified by opts
+func Consumer(stream string, cfg *ConsumerConfig) ConsumerOption {
 	return func(jopts *jsOpts) error {
 		jopts.consumer = cfg
```

Comment:
Should we process and fill in sane defaults here so we do not force users to fill in all of the fields?

Reply:
An API could allocate the consumer config struct and populate defaults  e.g. `NewConsumerConfig`.  It'd avoid issues around cases where an empty string or zero is a valid value but not necessarily the default or a good one.

Reply:
Yes, should have all the defaults there.

Reply:
It’s not needed the basic structure defaults inherently.